sharding/xorValidatorShuffler_test.go

```diff
@@ -918,6 +922,61 @@ func TestRandXORShuffler_UpdateNodeListsWithNonExistentLeavingDoesNotRemove(t *t
 	}
 }
 
+func TestRandXORShuffler_UpdateNodeListsWithRangeOnMaps(t *testing.T) {
+	t.Parallel()
+	shuffleBetweenShards := []bool{true, false}
+
+	for _, shuffle := range shuffleBetweenShards {
+		eligiblePerShard := 100
+		eligibleMeta := 100
+		shuffler := NewXorValidatorsShuffler(
+			uint32(eligiblePerShard),
+			uint32(eligibleMeta),
+			0.2,
+			false,
+		)
+		shuffler.shuffleBetweenShards = shuffle
+
+		waitingPerShard := 20
+		numShards := 2
+
+		numLeaving := (waitingPerShard + waitingPerShard/2) * (numShards + 1)
+
+		args := createShufflerArgs(eligiblePerShard, waitingPerShard, uint32(numShards))
+
+		allValidators := make([]Validator, 0)
+
+		for _, shardValidators := range args.Eligible {
+			allValidators = append(allValidators, shardValidators...)
+		}
+
+		for _, shardValidators := range args.Waiting {
+			allValidators = append(allValidators, shardValidators...)
+		}
+
+		leavingValidators := make([]Validator, numLeaving)
+
+		for i := 0; i < numLeaving; i++ {
+			randIndex := mathRand.Intn(len(allValidators))
+			leavingValidators[i] = allValidators[randIndex]
+			allValidators = removeValidatorFromList(allValidators, randIndex)
+		}
+
+		args.Leaving = leavingValidators
+
+		initialEligible, initialWaiting, initialLeaving := shuffler.UpdateNodeLists(args)
+
+		for i := 0; i < 100; i++ {
+			newArgs := copyArgs(args)
```

Comment:
Is this really necessary? Are we altering the provided slices from args?

Reply:
removed. the range problem seemed to happen every time if the maps were copied. But it also happens in 100 steps with the same map.